### ISSUE TITLE
impossible to install colorama

### ISSUE THREAD

**Issue by ericLemanissier** [tartley/colorama#240]
since few minutes, it seems to be impossible to install colorama:

```
pip3 install --upgrade colorama
Collecting colorama
  Using cached https://files.pythonhosted.org/packages/cb/fe/bfc4d807aa43a183ab387340f524a0bb086624f2c5935bd08e647b54b269/colorama-0.4.2.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-s_73iow0/colorama/setup.py", line 36, in <module>
        long_description=read_file('README.rst'),
      File "/tmp/pip-build-s_73iow0/colorama/setup.py", line 18, in read_file
        with open(os.path.join(os.path.dirname(__file__), path)) as fp:
    FileNotFoundError: [Errno 2] No such file or directory: '/tmp/pip-build-s_73iow0/colorama/README.rst'

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-s_73iow0/colorama/
```

**Comment by ellbosch** [comment]
I'm getting this error in Windows.

**Comment by vphilippon** [comment]
`README.rst` is somehow missing from the sdist (colorama-0.4.2.tar.gz), causing this error, and no wheel is available either.

**Comment by techied** [comment]
Having this issue with Ubuntu as well.

**Comment by podwhitehawk** [comment]
hm, master refering to v0.4.1
https://github.com/tartley/colorama/blob/master/colorama/__init__.py

**Comment by techied** [comment]
pypi shows 0.4.2 currently

**Comment by podwhitehawk** [comment]
no 0.4.2 tag either
https://github.com/tartley/colorama/tags

**Comment by BradenM** [comment]
Also experiencing this. 

Here is a log from tox running on the `windows-latest` env via github actions:

```
Collecting colorama<0.5.0,>=0.4.1; sys_platform == "win32"
  Downloading https://files.pythonhosted.org/packages/cb/fe/bfc4d807aa43a183ab387340f524a0bb086624f2c5935bd08e647b54b269/colorama-0.4.2.tar.gz
    ERROR: Command errored out with exit status 1:
     command: 'D:\a\micropy-cli\micropy-cli\.tox\py37\Scripts\python.EXE' -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\pip-install-jdcxa2nb\\colorama\\setup.py'"'"'; __file__='"'"'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\pip-install-jdcxa2nb\\colorama\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base 'C:\Users\RUNNER~1\AppData\Local\Temp\pip-install-jdcxa2nb\colorama\pip-egg-info'
         cwd: C:\Users\RUNNER~1\AppData\Local\Temp\pip-install-jdcxa2nb\colorama\
    Complete output (7 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\RUNNER~1\AppData\Local\Temp\pip-install-jdcxa2nb\colorama\setup.py", line 36, in <module>
        long_description=read_file('README.rst'),
      File "C:\Users\RUNNER~1\AppData\Local\Temp\pip-install-jdcxa2nb\colorama\setup.py", line 18, in read_file
        with open(os.path.join(os.path.dirname(__file__), path)) as fp:
    FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\pip-install-jdcxa2nb\\colorama\\README.rst'
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

[Link to full action](https://github.com/BradenM/micropy-cli/pull/75/checks?check_run_id=337352384)

**Comment by geerlingguy** [comment]
This has blown up all Travis CI builds for my open source projects too :(

**Comment by techied** [comment]
For now I suppose a workaround is to just use 0.4.1?

**Comment by qjflores** [comment]
this is breaking `pytest-watch`

**Comment by AXington** [comment]
This is a pretty big deal, it's breaking a lot of dependencies.

**Comment by ericLemanissier** [comment]
more precisely, the workaround is ` pip3 install --upgrade colorama==0.4.1 whatever_you_wanted_to_install`

**Comment by techied** [comment]
Unfortunately it is not always so simple to choose a different version

**Comment by geerlingguy** [comment]
It looks like `0.4.3` was just released a few minutes ago. My builds are succeeding again, yay!

https://github.com/tartley/colorama/releases/tag/0.4.3

**Comment by brherger** [comment]
Can the pypi release 0.4.2 be pulled until this is resolved?

**Comment by techied** [comment]
It should be resolved now according to @geerlingguy

**Comment by tartley** [comment]
This was all my fault. Apologies people.

I didn't realize the existence of the github tag was so critical - I'll go read about why. (clues appreciated)

and I don't know why the README.rst was missing from 0.4.2. Seems to be present in 0.4.3 though. I'll double check both the wheel and the sdist to be sure...

**Comment by AXington** [comment]
can confirm that this seems to be resolved now, however, I've changed my requirements.txt in several repos to include: `colorama>=0.4.1,!=0.4.2`

also @tartley not sure why/if that affected this last build, but typically tags are used to have a snapshot of releases so that a build can be reproduced if need be, or hotfixes applied, etc.

**Comment by tartley** [comment]
@AXington Thank you. I created the tag locally, just didn't push to github for a few minutes.

I tracked down some more diagnostic info about the missing README.rst:

This 0.4.2 sdist was built using my default python, 3.8, which produces an sdist without the README.rst. We have a makefile target to push both sdist and wheel to PyPI, and the sdist worked, but the wheel failed, so there was a period where there was no wheel, and in falling back to the sdist you saw the missing README file error.

I then built an 0.4.3 sdist using Python 3.6, which does include the README (I confirmed), and fixed the problem with wheel uploads.

**Comment by tartley** [comment]
I'll keep digging to find out all the details, so we can strive to avoid this happening again.

**Comment by tartley** [comment]
@wiggin15 When you wake:

1. Sorreeee! :-)
2. Extra my fault because I bypassed uploading to test.pypi.org (which I'll never do again) because I don't have privs to the 'colorama' package there. Can you remember how I get it? (ie how did we add you to 'colorama' on 'pypi.org'?)

**Comment by tartley** [comment]
3. Bloody deadlines. Was trying to get this published in time for the tidelift deadline, that ends today, and so I rushed it. Stupid.

**Comment by jrosser** [comment]
This broke openstack CI too - here is some really good advice about what to do with already released broken packages https://doughellmann.com/blog/2016/02/25/so-youve-released-a-broken-package-to-pypi-what-do-you-do-now/

0.4.2 will be in everyones mirrors for a long time now and won't go away

**Comment by nightlark** [comment]
Well, a new release that excludes Python 3.4 explains why our CI builds started failing on `pip3 install pytest`. I noticed that this release (unlike previous releases) says that it `Requires: Python >=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*` -- does colorama actually no longer work with Python 3.4, or was that requirement added for other reasons (e.g. deprecating support for older Python versions)?

**Comment by wiggin15** [comment]
> 2. Extra my fault because I bypassed uploading to test.pypi.org (which I'll never do again) because I don't have privs to the 'colorama' package there. Can you remember how I get it? (ie how did we add you to 'colorama' on 'pypi.org'?)

@tartley  I added you as an owner of the colorama project on `test.pypi.org` (apparently I was the "sole owner". I don't know why).

**Comment by jongiddy** [comment]
Also, please update the `CHANGELOG.rst` to indicate the changes for both new versions, so we can see what's new in 0.4.2 and why the fix in 0.4.3 was released on the same day.

**Comment by jayvdb** [comment]
Related issue: https://github.com/tartley/colorama/issues/243

**Comment by wiggin15** [comment]
Original problem was fixed by release 0.4.3.
Python 3.4 is EOL and was intentionally removed.

**Comment by theoturner** [comment]
Could not install or uninstall 0.4.4. Just go to your site packages and remove any colorama directories, then reinstall.
